jax/_src/numpy/lax_numpy.py

```diff
@@ -5373,7 +5373,7 @@ def digitize(x: ArrayLike, bins: ArrayLike, right: bool = False) -> Array:
   if bins_arr.ndim != 1:
     raise ValueError(f"digitize: bins must be a 1-dimensional array; got {bins=}")
   if bins_arr.shape[0] == 0:
-    return zeros(x, dtype=dtypes.canonicalize_dtype(int_))
+    return zeros(x.shape, dtype=dtypes.canonicalize_dtype(int_))
```

Comment:
SInce `x` is `ArrayLike`, it doesn't necessarily have a `shape` attribute. I think `return zeros_like(x, dtype=...)` should do the right thing.